migrate.ts

```diff
@@ -4,53 +4,76 @@
  * If no environment is specified, you can use the environment variables PG_URL
  * and SETTINGS_FILE
  */
-require("ts-node/register");
-const { getDatabaseConfig, startSshTunnel } = require("./scripts/startup/buildUtil");
-
-const initGlobals = (args, isProd) => {
-  global.bundleIsServer = true;
-  global.bundleIsTest = false;
-  global.bundleIsE2E = false;
-  global.bundleIsProduction = isProd;
-  global.bundleIsMigrations = true;
-  global.defaultSiteAbsoluteUrl = "";
-  global.serverPort = 5001;
-  global.estrellaPid = -1;
+
+import { existsSync } from "node:fs";
+import type { ITask } from "pg-promise";
+
+// @ts-ignore This is a javascript file without a .d.ts
+import { getDatabaseConfig, startSshTunnel } from "./scripts/startup/buildUtil";
+
+const initGlobals = (args: Record<string, unknown>, isProd: boolean) => {
+  Object.assign(global, {
+    bundleIsServer: true,
+    bundleIsTest: false,
+    bundleIsE2E: false,
+    bundleIsProduction: isProd,
+    bundleIsMigrations: true,
+    defaultSiteAbsoluteUrl: "",
+    serverPort: 5001,
+    estrellaPid: -1,
+  });
 
   const { getInstanceSettings } = require("./packages/lesswrong/lib/executionEnvironment");
   getInstanceSettings(args); // These args will be cached for later
 }
 
-const fetchImports = (args, isProd) => {
+const fetchImports = (args: Record<string, unknown>, isProd: boolean) => {
   initGlobals(args, isProd);
 
   const { getSqlClientOrThrow, setSqlClient } = require("./packages/lesswrong/server/sql/sqlClient");
   const { createSqlConnection } = require("./packages/lesswrong/server/sqlConnection");
   return { getSqlClientOrThrow, setSqlClient, createSqlConnection };
 }
 
-const credentialsPath = (forumType) => {
+type ForumType = "lw" | "ea" | "none";
+
+const detectForumType = async (): Promise<ForumType> => {
+  const base = process.env.GITHUB_WORKSPACE ?? "..";
+  if (existsSync(`${base}/LessWrong-Credentials`)) {
+    return "lw";
+  }
+  if (existsSync(`${base}/ForumCredentials`)) {
```

Comment:
Possibly the dirnames here should use `memorizedRepoNames` (which can be pulled out of the `credentialsPath` function)?